### ISSUE TITLE
Relax error FS0208: --target:exe require .exe extension

### ISSUE THREAD

**Issue by enricosada** [Microsoft/visualfsharp#824]
if i try to compile with 

```fsc --target:exe Program.fs --out:prova.notexe```

i get error FS0208

```
error FS0208: The output name extension doesn't match the options used. 
If '-a' or '--target:library' is used the output file name must end with '.dll', 
if '--target:module' is used the output extension must be '.netmodule', 
otherwise '.exe'.
```

that's a problem because dotnet cli ( the old dnx ) compile to `.dll` instead of .exe

Is possibile to remove this error completly? We can disable it ( with `--nowarn` maybe ) but is not really useful

bonus point: remove it from coreclr branch first, so i can remove the workaround from dotnet cli

**Comment by KevinRansom** [comment]
Yes we can remove it from the compiler, it is not really a useful restriction.

**Comment by KevinRansom** [comment]
I will take care of it over Christmas.

**Comment by enricosada** [comment]
good :smile:

**Comment by ProTip** [comment]
I get an error on this with the latest dotnet CLI on linux. Can't wait for Ionide, F#, and friends to be stable for CoreCLR on linux(C# for that matter too).

**Comment by enricosada** [comment]
@ProTip what version of cli ( `dotnet --version` or `dotnet --info` if supported )

**Comment by ProTip** [comment]
```
$ dotnet --version
.NET Command Line Tools (1.0.0-beta-001793)

Product Information:
 Version:     1.0.0-beta-001793
 Commit Sha:  42a0eec967

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 Runtime Id:  ubuntu.14.04-x64
```

**Comment by ProTip** [comment]
I could be throwing you for a loop here(unrelated issue). This is what I get:

```
$ dotnet build
Compiling fsharp for .NETStandardApp,Version=v1.5
/usr/share/dotnet/bin/dotnet compile-fsc @/home/gzapp/projects/fsharp/obj/Debug/netstandardapp1.5/dotnet-compile.rsp returned Exit Code 1
/home/gzapp/projects/fsharp/error FS0208: The output name extension doesn't match the options used. If '-a' or '--target:library' is used the output file name must end with '.dll', if '--target:module' is used the output extension must be '.netmodule', otherwise '.exe'.
```

**Comment by enricosada** [comment]
in that version of .net cli, f# doesnt support `netstandardapp1.5`, and it's trying to compile without `--profile: netcore`

if you do `dotnet --verbose build` you can see arguments ( or check .rsp file in `obj/Debug/netstandard1.5/dotnet-compile-fsc.rsp` )

**You can try version version `1.0.0-beta-002071` **, that should work ( see https://github.com/enricosada/fsharp-dotnet-cli-samples/wiki/Status )

**Comment by dsyme** [comment]
can we close this?

**Comment by enricosada** [comment]
yes @dsyme, i see it working locally (.net core preview2, so latest compiler).

i think the fix is in commit 79165ef4318e30bb81200ecb910660aa28a2b733 in pr #1159
